server/v2/api/rest/server.go

```diff
@@ -89,6 +87,12 @@ func (s *Server[T]) Stop(ctx context.Context) error {
 	}
 
 	s.logger.Info("stopping HTTP server")
+	defer func() {
```

Comment:
If `httpServer` is set in `New` how come we need this?

Reply:
If we call `Start` again later after `Stop`, the old httpServer will fail since Shutdown was called.

Reply:
I see, but why would you do that? That's not the contract.
I think we should just add in the server component docs, like in the std http server docs something like:

`Once Stop has been called on a server, it may not be reused`.

Reply:
Before we wanted to support server reloading, so then it would have made sense, but due to the config reloading limitation, we haven't gone that way. I think it makes server components simpler to write if you know once you call stop you don't need to make it re-startable.